src/treeffuser/_base_tabular_diffusion.py

```diff
@@ -97,41 +99,137 @@ def get_new_score_model(self) -> ScoreModel:
+                        X[c].dtype, np.number
+                    ):
+                        raise ValueError(
+                            f"Dtypes of columns in `X` must be int, float, bool or category. "
+                            f"Column {c} has dtype {X[c].dtype}."
+                            f"Convert the column to a numerical dtype or a category, e.g.,"
+                            f"X['{c}'] = X['{c}'].astype('category')."
+                        )
+                    if isinstance(X[c].dtype, pd.CategoricalDtype):
+                        if X_has_been_copied is False:
+                            # copy the DataFrame to avoid modifying the original
+                            X = X.copy()
+                            X_has_been_copied = True
+                        X[c] = X[c].cat.codes
+                # at that point, X contains only numerical columns
+                if reset:
+                    # store the columns of the DataFrame
+                    self._x_dataframe_columns = X.columns
+                    self._x_cat_idx = cat_idx
+                    self._x_dim = X.shape[1]
+                else:
+                    # check coherence of the input data with the training data
+                    # ensure the columns of the DataFrame are the same as the ones used during training
+                    if not all(col in X.columns for col in self._x_dataframe_columns):
+                        raise ValueError(
+                            "Column names in `X` are not present in the DataFrame but were present during training."
+                            f"{set(self._x_dataframe_columns) - set(X.columns)}"
+                        )
+                    # reorder the columns of the DataFrame to match the order of the training data
+                    X = X[self._x_dataframe_columns]
+                X = X.values
+
+            elif isinstance(X, np.ndarray):
+                if cat_idx is not None:
+                    for idx in cat_idx:
+                        if idx < 0 or idx >= X.shape[1]:
+                            raise ValueError(
+                                f"Invalid indices in `cat_idx`: {idx} is not in "
+                                f"[0, {X.shape[1]}-1] (the shape of X)."
+                            )
+                else:
+                    cat_idx = []
+
+                if reset:
+                    self._x_cat_idx = cat_idx
+            else:
+                raise TypeError(
+                    "Input data `X` must be a numpy array or a pandas DataFrame."
+                    f"Reveived {type(X).__name}."
+                )
+
             X = _check_array(X)
 
-        if validate_y and y is not None:
+        if validate_y:
+            if y is None:
+                raise ValueError("Target data `y` cannot be None.")
+            if isinstance(y, pd.Series):
+                y = y.values
+            elif not isinstance(y, np.ndarray):
```

Comment:
why does it need to be a series. Couldn't we have a multivariate y?

Reply:
oops good catch thanks!